controllers/scopetemplate_controller_test.go

```diff
@@ -0,0 +1,548 @@
+					}))
+					Expect(existingRB.RoleRef).To(Equal(rbacv1.RoleRef{
+						Kind:     "ClusterRole",
+						Name:     "test",
+						APIGroup: "rbac.authorization.k8s.io",
+					}))
+				})
+
+				When("a scopeInstance is updated to remove one of the namespace", func() {
+
+					It("Should remove respective roleBindings in the expected namespaces", func() {
+						Eventually(func() error {
+							scopeInstance := &operatorsv1.ScopeInstance{}
+							Expect(k8sClient.Get(ctx, types.NamespacedName{Name: "scopeinstance-sample"}, scopeInstance)).NotTo(HaveOccurred())
+							scopeInstance.Spec.Namespaces = []string{namespace2.GetName()}
+							Expect(k8sClient.Update(ctx, scopeInstance)).NotTo(HaveOccurred())
+							return nil
+						}, timeout, interval).Should(BeNil())
+
+						roleBindingList := &rbacv1.RoleBindingList{}
+						Eventually(func() error {
+							err := k8sClient.List(ctx, roleBindingList, &client.ListOptions{
+								Namespace: namespace2.GetName(),
+							},
+								client.MatchingLabels{
+									scopeInstanceUIDKey:           string(scopeInstance.GetUID()),
+									scopeTemplateUIDKey:           string(scopeTemplate.GetUID()),
+									clusterRoleBindingGenerateKey: "test",
+								})
+							if err != nil {
+								return err
+							}
+
+							if len(roleBindingList.Items) != 1 {
+								return fmt.Errorf("Expected 1 roleBinding, found %d", len(roleBindingList.Items))
+							}
+
+							return nil
+						}, timeout, interval).Should(BeNil())
+
+						existingRB := &roleBindingList.Items[0]
+
+						Expect(len(existingRB.OwnerReferences)).To(Equal(1))
+						Expect(existingRB.OwnerReferences).Should(ContainElement(metav1.OwnerReference{
+							APIVersion: "operators.io.operator-framework/v1",
+							Kind:       "ScopeInstance",
+							Name:       scopeInstance.GetObjectMeta().GetName(),
+							UID:        scopeInstance.GetObjectMeta().GetUID(),
+						}))
+
+						Expect(len(existingRB.Subjects)).To(Equal(1))
+						Expect(existingRB.Subjects).Should(ContainElement(rbacv1.Subject{
+							Kind:     "Group",
+							Name:     "manager",
+							APIGroup: "rbac.authorization.k8s.io",
+						}))
+						Expect(existingRB.RoleRef).To(Equal(rbacv1.RoleRef{
+							Kind:     "ClusterRole",
+							Name:     "test",
+							APIGroup: "rbac.authorization.k8s.io",
+						}))
+
+						roleBindingList = &rbacv1.RoleBindingList{}
+						Eventually(func() error {
+							err := k8sClient.List(ctx, roleBindingList, &client.ListOptions{
+								Namespace: namespace.GetName(),
+							},
+								client.MatchingLabels{
+									scopeInstanceUIDKey:           string(scopeInstance.GetUID()),
+									scopeTemplateUIDKey:           string(scopeTemplate.GetUID()),
+									clusterRoleBindingGenerateKey: "test",
+								})
+							if err != nil {
+								return err
+							}
+							return nil
+						}, timeout, interval).Should(BeNil())
+
+						Expect(len(roleBindingList.Items)).To(Equal(0))
```

Comment:
If the list call is made before the roleBinding is cleaned up, the test will fail, you need to check the length of the list within the Eventually func.

```suggestion
						Eventually(func() error {
							err := k8sClient.List(ctx, roleBindingList, &client.ListOptions{
								Namespace: namespace.GetName(),
							},
								client.MatchingLabels{
									scopeInstanceUIDKey:           string(scopeInstance.GetUID()),
									scopeTemplateUIDKey:           string(scopeTemplate.GetUID()),
									clusterRoleBindingGenerateKey: "test",
								})
							if err != nil {
								return err
							}
							
							if len(roleBidningList.Items) != 0 {
								return fmt.Errorf("Expected 0 roleBindings, found %d", len(roleBindingList.Items))
							}
							return nil
						}, timeout, interval).Should(BeNil())

```

Reply:
Modified and agree with it.